lib/boom_notifier/notifier.ex

```diff
@@ -4,4 +4,6 @@ defmodule BoomNotifier.Notifier do
   """
 
   @callback notify(list(%ErrorInfo{}), keyword(String.t())) :: no_return()
+  @callback validate!(keyword(String.t())) :: no_return()
```

Comment:
Thinking on better communicating the intention of the callback... what about naming it `validate_configuration` or `validate_config`?